### PR TITLE
checkpoint fat

### PR DESCRIPTION
**Pull request by sebratton**

### CODE REVIEW

**Comment by sebratton** [comment]
#build

**Comment by sebratton** [comment]
#build

File: dev/fattest.simplicity/src/componenttest/topology/impl/LibertyServer.java [hunk 1]
```diff
@@ -450,10 +452,76 @@ public String getPathToAutoFVTNamedServer() {
...
+     * <code> "bin/server start"</code>
+     * </pre>
+     *
+     * @param phase
+     * @param autoRestor    if true initiate restore as part of serverStart
+     * @param beforeRestore beforeRestore lambda is called just before the server start
      */
-    private Phase checkpointPhase = null;
+    public void setCheckpoint(Phase phase, boolean autoRestore, Consumer<LibertyServer> beforeRestoreLambda) {
+        if (phase == null) {
```
Comment (tjwatson) [inline dev/fattest.simplicity/src/componenttest/topology/impl/LibertyServer.java]: Why check for null here but not in the other setCheckpoint method above?  Why doesn't the simple `setCheckpoint(Phase phase)` method just call this one with `true` and a `null` consumer?

**Review by tjwatson — APPROVED** [review]
Just a small comment on the constructor.  Overall it seems like the right direction.

File: dev/fattest.simplicity/src/componenttest/topology/impl/LibertyServer.java (hunk 1 above)
Comment (sebratton) [reply dev/fattest.simplicity/src/componenttest/topology/impl/LibertyServer.java]: Thanks. I moved the null check to constructor and redirected one parameter setCheckpoint method to call three parameter version.

**Comment by sebratton** [comment]
#build

**Comment by sebratton** [comment]
#build

**Comment by sebratton** [comment]
#build

**Pull request merged**